components/root/root.jsx

```diff
@@ -207,6 +207,10 @@ export default class Root extends React.Component {
             this.props.history.push('/get_android_app?redirect_to=' + encodeURIComponent(this.props.location.pathname) + encodeURIComponent(this.props.location.search));
             BrowserStore.setLandingPageSeen(true);
         }
+
+        if (this.props.noAccounts) {
```

Comment:
This was previously done in `afterIntl` ([link](https://github.com/mattermost/mattermost-webapp/blob/release-5.5/components/root/root.jsx#L204)), so perhaps we should do the same here? This flow is a mess, so I'm hesitant to move where this is done

Reply:
I'm not opposed to moving it into `afterIntl` but that will make unit testing difficult. I don't want to call `redirectIfNecessary` again because I'm not confident that won't cause issues with the TOS feature and because that function has changed since 5.5

Reply:
^ @hmhealey